pkg/storage/snapshot/restore.go

```diff
@@ -596,6 +606,115 @@ func (t *vmRestoreTarget) reconcileDataVolumes() (bool, error) {
 	return createdDV || waitingDV, nil
 }
 
+func (t *vmRestoreTarget) getControllerRevision(namespace, name string) (*appsv1.ControllerRevision, error) {
+	revisionKey := cacheKeyFunc(namespace, name)
+	obj, exists, err := t.controller.CRInformer.GetStore().GetByKey(revisionKey)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("Unable to find ControllerRevision %s", revisionKey)
+	}
+	return obj.(*appsv1.ControllerRevision), nil
+}
+
+func (t *vmRestoreTarget) getVirtualMachineSnapshot(namespace, name string) (*snapshotv1.VirtualMachineSnapshot, error) {
+	vmSnapshotKey := cacheKeyFunc(namespace, name)
+	obj, exists, err := t.controller.VMSnapshotInformer.GetStore().GetByKey(vmSnapshotKey)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("Unable to find VirtualMachineSnapshot %s", vmSnapshotKey)
+	}
+	return obj.(*snapshotv1.VirtualMachineSnapshot), nil
+}
+
+func (t *vmRestoreTarget) restoreInstancetypeControllerRevision(revisionName, vmSnapshotName string, vm *kubevirtv1.VirtualMachine) (*appsv1.ControllerRevision, error) {
+	existingCR, err := t.getControllerRevision(vm.Namespace, revisionName)
+	if err != nil {
+		return nil, err
+	}
+
+	vmSnapshot, err := t.getVirtualMachineSnapshot(vm.Namespace, vmSnapshotName)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the CR is controlled by the VirtualMachineSnapshot restore it and attempt to claim it if the VirtualMachine exists
+	if metav1.IsControlledBy(existingCR, vmSnapshot) {
```

Comment:
Isn't this always the case?
IIUC (in your first commit) the snapshot controller would clone the CR and unconditionally set it to be the source's CR.

Reply:
It should be yes but I wanted to be sure we were working on the current CR so this seemed like a valid check.

Reply:
IMHO it should be clear from the code, as someone without context would have hard time figuring this out.

Is there a reason in your opinion we should return an error if `metav1.IsControlledBy(existingCR, vmSnapshot)` is false?
Either way, if not an error, I think a warning log entry is necessary here.

Reply:
ACK as discussed I've dropped this check.